### PR TITLE
feat: bump to version v2.4

### PR DESCRIPTION
**Pull request by starsz**
Please answer these questions before submitting a pull request

- Why submit this pull request?

for version 2.4 release

### CODE REVIEW

**Review by imjoey — APPROVED** [review]
Everything looks good to me, only one typo.

File: CHANGELOG.md [hunk 1]
```diff
@@ -27,6 +28,43 @@
 - [1.5.0](#150)
 - [1.0.0](#100)
 
+# 2.4.0
+
+This release mainly improve basic features, bugfix and adds test cases.
```
[the remark this answers is not in the document]
Comment (starsz) [reply CHANGELOG.md:33]: Ah, thanks very much.

File: CHANGELOG.md [hunk 2]
```diff
@@ -27,6 +28,43 @@
...
+### Core
+
+- Support to show warning notification when dashboard version not matching apisix. [#1435](https://github.com/apache/apisix-dashboard/pull/1435)
+- Support mTLS connection to ETCD. [#1437](https://github.com/apache/apisix-dashboard/pull/1437)
+- Support IP allow list. [#1424](https://github.com/apache/apisix-dashboard/pull/1424)
+- Support to get manager-api version through API. [#1429](https://github.com/apache/apisix-dashboard/pull/1429)
+- Support import route from OpenAPI specification3.0. [#1102](https://github.com/apache/apisix-dashboard/pull/1102)
+- Support export route from OpenAPI specification3.0. [#1245](https://github.com/apache/apisix-dashboard/pull/1245)
+- Support string type for the script field in Route. [#1289]((https://github.com/apache/apisix-dashboard/pull/1245)
+- Feat: add script_id field in Route.entity. [#1386](https://github.com/apache/apisix-dashboard/pull/1386)
```
Comment (juzhiyuan) [inline CHANGELOG.md:46]: `feat` is also like `support`, right?

**Review by nic-chen — COMMENTED** [review]
should run `./utils/version-check.sh 2.4`  to check version and modify accordingly

**Review by juzhiyuan — APPROVED** [review]

File: CHANGELOG.md (hunk 2 above)
Comment (starsz) [reply CHANGELOG.md:46]: Right

**Pull request merged**